modules/apps/27-interchain-accounts/types/codec.go

```diff
@@ -26,14 +26,18 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 }
 
 // SerializeCosmosTx serializes a slice of sdk.Msg's using the CosmosTx type. The sdk.Msg's are
-// packed into Any's and inserted into the Messages field of a CosmosTx. The proto marshaled CosmosTx
-// bytes are returned. Only the ProtoCodec is supported for serializing messages.
-func SerializeCosmosTx(cdc codec.BinaryCodec, msgs []proto.Message, encoding string) (bz []byte, err error) {
-	// only ProtoCodec is supported
+// packed into Any's and inserted into the Messages field of a CosmosTx. The CosmosTx is marshaled
+// depending on the encoding type passed in. The marshaled bytes are returned. Only the ProtoCodec
+// is supported for serializing messages. Both protobuf and proto3 JSON are supported.
+func SerializeCosmosTx(cdc codec.Codec, msgs []proto.Message, encoding string) ([]byte, error) {
+	// this is a defensive check to ensure only the ProtoCodec is used for message serialization
 	if _, ok := cdc.(*codec.ProtoCodec); !ok {
-		return nil, errorsmod.Wrap(ErrInvalidCodec, "only ProtoCodec is supported for receiving messages on the host chain")
+		return nil, errorsmod.Wrap(ErrInvalidCodec, "only the ProtoCodec may be used for receiving messages on the host chain")
 	}
 
+	var bz []byte
+	var err error
```

Comment:
I'm surprised the linter isn't complaining about this now being:

```go
var (
    bz []byte
    err error
)
```

I swear there was once another linter before which enforced that! Doesn't bother me either way